src/main/java/com/google/sps/servlets/ProfileServlet.java

```diff
@@ -41,52 +44,77 @@
       System.err.println(ErrorMessages.TEMPLATES_DIRECTORY_NOT_FOUND);
     }
-    if (!userService.isUserLoggedIn()) {
-      response.sendRedirect("/landing");
-    } else {
-      String userId = userService.getCurrentUser().getUserId();
-      Query query =
-          new Query("UserAccount")
-              .setFilter(
-                  new Query.FilterPredicate("userID", Query.FilterOperator.EQUAL, requestedUserID));
-      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-      PreparedQuery result = datastore.prepare(query);
-      Entity userEntity = result.asSingleEntity();
-      Map<String, Object> context = new HashMap();
-      context.put("url", "/profile?userID=" + requestedUserID);
-      context.put("userType", (long) (userEntity.getProperty("userType")));
-      context.put("browsingUserProfileURL", "/profile?userID=" + userId);
-      if ((long) (userEntity.getProperty("userType")) == 0) {
-        context.put("mentor", new Mentor(userEntity));
-      } else {
-        context.put("mentee", new Mentee(userEntity));
-      }
+
+    Map<String, Object> context = new HashMap<>();
+
+    try {
       String template =
           Resources.toString(
               this.getClass().getResource(ResourceConstants.TEMPLATE_PROFILE), Charsets.UTF_8);
-      String renderedTemplate = jinjava.render(template, context);
-      response.setContentType("text/html;");
-      response.getWriter().println(renderedTemplate);
+      profileTemplate = jinjava.render(template, context);
+    } catch (IOException e) {
+      System.err.println(ErrorMessages.templateFileNotFound(ResourceConstants.TEMPLATE_PROFILE));
     }
   }
 
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    User user = dataAccess.getCurrentUser();
+    if (user == null) {
+      response.sendRedirect(URLPatterns.LANDING);
+      return;
+    }
+    String userID = user.getUserId();
+
+    UserAccount userAccount = dataAccess.getUser(userID);
+    if (userAccount == null) {
+      response.sendRedirect(URLPatterns.LANDING);
+      return;
+    }
+
+    String requestedUserID = getParameter(request, "userID", userID);
+    Query query =
+        new Query(UserAccount.ENTITY_TYPE)
+            .setFilter(
+                new Query.FilterPredicate(
+                    ParameterConstants.USER_ID, Query.FilterOperator.EQUAL, requestedUserID));
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery result = datastore.prepare(query);
+    Entity userEntity = result.asSingleEntity();
+    Map<String, Object> context = dataAccess.getDefaultRenderingContext(URLPatterns.PROFILE);
+    context.put("userType", (long) (userEntity.getProperty("userType")));
+    context.put("browsingUserProfileURL", "/profile?userID=" + userID);
+    if ((long) (userEntity.getProperty("userType")) == 0) {
+      context.put("mentor", new Mentor(userEntity));
+    } else {
```

Comment:
This should be an elseif so that you are certain it is actually the mentee type. Otherwise it will break if you add more user types.